### PR TITLE
move NegStatus from pkg/neg to pkg/annotations

### PR DESCRIPTION
**Pull request by freehan**
This aims to make it easier to vendor pkg/annotations from a different repo

### CODE REVIEW

**Comment by freehan** [comment]
cc @spencerhance @cadmuxe

File: pkg/neg/types/types.go [hunk 1]
```diff
@@ -80,39 +79,10 @@ func (p1 PortInfoMap) Difference(p2 PortInfoMap) PortInfoMap {
 	return result
 }
 
-func (p1 PortInfoMap) ToPortNegMap() PortNegMap {
-	ret := PortNegMap{}
+func (p1 PortInfoMap) ToPortNegMap() map[string]string {
```
Comment (MrHohn) [inline pkg/neg/types/types.go]: Why not using `annotations.PortNegMap` here?

Comment (freehan) [reply pkg/neg/types/types.go]: I was trying to trim dependency of pkg/neg. But seems not possible. will revert it back

**Comment by freehan** [comment]
Updated!

**Pull request merged**